### ISSUE TITLE
bysetpos with multiple byweekday in monthly recurrence giving unexpected results

### ISSUE THREAD

**Issue by VinitaPotter** [jakubroztocil/rrule#375]
When the recurrence rule is set for every month on 3rd Monday and Wednesday, the resulting dates are messed up. Attaching a picture from the demo site.

new RRule({
  freq: RRule.MONTHLY,
  count: 30,
  interval: 1,
  byweekday: [RRule.MO, RRule.WE],
  bysetpos: [3]
})

String: RRULE:FREQ=MONTHLY;COUNT=30;INTERVAL=1;WKST=MO;BYDAY=MO,WE;BYSETPOS=3

But the resulting text and the result are incorrect

rule.toText(): _every month on Monday, Wednesday for 30 times_
--
expected rule.toText(): _every month on 3rd Monday and Wednesday._

**NOTE**: bysetpos DOES work with one byweekday but not multiple days array

![RRule bug](https://user-images.githubusercontent.com/36228361/69859245-c6b19d00-12b9-11ea-870b-7d586761fcfe.png)

**Comment by Mattkins** [comment]
Can confirm I'm seeing the same issue in my own implementation. Is the preference that we use the .nth method instead of trying to combine setpost and weekday?

**Comment by Mattkins** [comment]
For anyone else looking here, I worked around this using the .nth method on the weekday constants.

`          rule.freq = RRule.MONTHLY
          rule.byweekday = []
          this.formGroup.value.byWeekDay.forEach((weekDay) => {
            this.formGroup.value.bySetPos.forEach((setPos) => {
              rule.byweekday.push(weekDay.nth(setPos))
            })
          })`

**Comment by tanthammar** [comment]
@Mattkins where does the .nth() method come from?

**Comment by Mattkins** [comment]
Hopefully I don't lead you on a wild goose chase, it's been a while and we don't work with this library any more. I believe it's on the weekday constants provided by RRule. For example:

`const firstTuesday = RRule.TU.nth(1)`

**Comment by tanthammar** [comment]
> Hopefully I don't lead you on a wild goose chase, it's been a while and we don't work with this library any more. I believe it's on the weekday constants provided by RRule. For example:
> 
> `const firstTuesday = RRule.TU.nth(1)`

Yes, I found it. Thank you for replying. May I ask which library you are using instead?

**Comment by Mattkins** [comment]
@tanthammar, we don't do anything with rrule or any repeating dates any more, sorry.
